provider/target_server_group_stage.go

```diff
@@ -0,0 +1,363 @@
+package provider
+
+import (
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/jgramoll/terraform-provider-spinnaker/client"
+)
+
+type targetServerGroupStage struct {
+	// baseStage
+	Name                              string                   `mapstructure:"name"`
+	RefID                             string                   `mapstructure:"ref_id"`
+	Type                              client.StageType         `mapstructure:"type"`
+	RequisiteStageRefIds              []string                 `mapstructure:"requisite_stage_ref_ids"`
+	Notifications                     *[]*notification         `mapstructure:"notification"`
+	StageEnabled                      *[]*stageEnabled         `mapstructure:"stage_enabled"`
```

Comment:
I think this is likely from the existing code, but pointer to slice isn't super recommended as a slice is already essentially a pointer.

Reply:
Yes, he simply moved this to an abstraction letting him reuse it. You also need to take care of those code refactoring to avoid merging conflicts with the official repo. Should we also send it to the official repo after this merging?